### DISCUSSION TITLE
Suggestion: service to rename area id

### DISCUSSION THREAD

**Discussion by tjorim** [frenck/spook#208] [Ideas]
Currently area id's cannot be changed easily.

If you initially made a typo in the name or you created them in your native language and want to switch to English, there is no way to do that.

You can rename an area and add aliases, but in your yaml automations for example, you still need to use the old area id.

Zones can be changed without a problem, not the case for areas. I once tried it by manually changing the config storage but borked my whole installation that way.

**Comment by frenck** [comment]
I moved the issue into a discussion.

> Zones can be changed without a problem, not the case for areas.

Correct, they are fundamentally different. Zone is an entity, an area is not.

The universal solution is to use the `area_id()` template method:

```jinja2
target:
  area_id: "{{ area_id("Woonkamer") }}"
```

In this case, you'd always use the normal name and the ID isn't relevant (it should be relevant).

> Suggestion: service to rename area id 

This is a hard one. We could make a service that updates the registry, but as you stated and learned, that is not enough. We should update all referring registries as well (devices & entities). Then there are automation and dashboards, which may or may not, be UI-managed.
